### DISCUSSION TITLE
Restructure variants of OV Images

### DISCUSSION THREAD

**Discussion by linuxianer99** [Openvario/meta-openvario#181] [Ideas]
This should be a general discussion about the needed types of images and their content of tools.

As as starting point, we can see 3 types of users:

- Normal user: Just using the ready image to flash it on the OV and go flying ..
- Power user: Uses the OV for flying, but wants to improve or change some configuration items not listed in the menus (no heavy programming skills)
- Developer: Like the OV so much in flying, that he also wants to contribute a lot (heavy programming skills)

Based on the users, we should discuss the content and variants of the images .....

**Comment by mihu-ov** [comment]
From my experience in our club most users are "normal users" but need help from a "power user" from time to time because something went wrong, some file is missing, the USB stick failed and the other one has no OV directory structure, etc.
Unless there is a major disadvantage I suggest keeping these two images the same and the tools we have in.

**Comment by sascha-daehne** [comment]
One point is the connectivity. In former versions bluetooth was available. Discussions about the minimum image size leads to missing hci tools, which are necessary for bluetooth connection.
One Use-Case is an LXNANO Logger, which is connected to my OV via bluetooth.
I just managed to compile an image by myself, but I have no idea where to define packages, which should included into the image. Is it further possible to add them afterwords?
So from my point of view this connectivity issue can be also relevant for "Normal Users". But maybe only if they are using a LXNANO.
I agree with @mihu-ov who mentioned to have one version for normal and power users. The developers have the very nice build environment, where they can do whatever they want to. So I see only ONE release image.

**Reply by MaxKellermann** [reply]
> Discussions about the minimum image size leads to missing hci tools, which are necessary for bluetooth connection.

This is wrong on all levels. If something depends on hci-tools, it will be included in the image. If nothing depends on hci-tools, then it will be missing. The real question is: does anything depend on hci-tools? If yes, then your problem does not exist. If not, what should be?

**Reply by sascha-daehne** [reply]
Ok, the reason for omitting hcitools was an assumption, so I'm wrong in this case. hci-tools has no direct dependency, but I need it to couple with my lxnano logger over bluetooth. Then the question is: should bluetooth connections should be available with OV? 
If so (And I vote for supporting bluetooth connectivity in the release image) hci-tools is a (soft) dependency. 
(Maybe there are other solutions I'm not aware of, but hci-tools is one solution working before)

**Comment by lordfolken** [comment]
There is no hard dependency as the common interface is serial ports.
But without it you can't add Bluetooth based devices to xcsoar. Almost every current generation vario (lxnav, lxnavigation, xcvario...) offers WiFi and Bluetooth connectivity.
I don't see an advantage in making the enduser jump through extra hoops by not providing basic connectivity tools in the image.
I would vote for adding a dependency on hci-tools in the xcsoar package in meta-ov.
